### PR TITLE
Added local to the local variables

### LINKED ISSUE #5: Use local for variable declaration in functions
Hi, i noticed that the script defines values in the functions without using the `local` declaration.

See: [linux.die.net#23.2. Local Variables](https://linux.die.net/abs-guide/localvar.html).

Using `local` prevents weird bugs, when the same variable is defined in different functions.

### PR DESCRIPTION
**Pull request by wargio**
Fixes #5

### CODE REVIEW

**Comment by d3jk** [comment]
@wargio This is great! Thanks for contributing! We'll review and test and hopefully merge as soon as we can!

File: pre-receive.sh [hunk 1]
```diff
@@ -202,7 +202,7 @@ function EXIT_SEDATED() {
 # Check if a line contains a filename from patch file
 # $1 is the line to be checked
 function CHECK_IF_LINE_CONTAINS_FILENAME() {
-  temp_filename=$(echo "$1" | grep -Po '^\+\+\+ b/\K(.*)')
+  local temp_filename=$(echo "$1" | grep -Po '^\+\+\+ b/\K(.*)')
```
Comment (d3jk) [inline pre-receive.sh]: @wargio could you remove the local variable assignment from this line (205)? This variable is intentionally global.

**Review by d3jk — CHANGES_REQUESTED** [review]
@wargio please see comment on line 205 of pre-receive.sh

File: pre-receive.sh (hunk 1 above)
Comment (wargio) [reply pre-receive.sh]: but that variable `temp_filename` is used only there!

Comment (d3jk) [reply pre-receive.sh]: Checkout line 422 in that file. It is used in the `if` statement.
`if [[ ! "$temp_filename" ]]; then`

Comment (wargio) [reply pre-receive.sh]: looks like my fork was older. i'll remove the local.

**Review by d3jk — COMMENTED** [review]
Looks good. We'll review/test with these changes and hopefully get this merged in soon! Thanks for contributing! :)

**Review by d3jk — APPROVED** [review]

**Pull request merged**